src/panels/config/automation/ha-automation-editor.ts

```diff
@@ -257,6 +295,32 @@ class HaAutomationEditor extends LitElement {
     );
   }
 
+  private _toggleYaml(): void {
+    this._showYaml = !this._showYaml;
+  }
+
+  private get _yamlEditor(): HuiYamlEditor | null {
+    return this.shadowRoot!.querySelector("hui-yaml-editor");
+  }
+
+  private get _yaml(): string {
+    if (this._config) {
+      return yaml.safeDump(this._config);
+    }
+    return "";
+  }
+
+  private _handleYamlChanged(ev): boolean {
+    ev.stopPropagation();
+    try {
+      this._config = yaml.safeLoad(ev.detail.value);
+      this._dirty = true;
+      return true;
+    } catch (exception) {
+      return false;
```

Comment:
We should show the error to the user, instead of just not converting the yaml.

Reply:
Now from the moment the user has made an error in the yaml, all his changes are ignored. You should not be allowed to close the yaml editor if there are yaml errors.